plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -584,7 +585,9 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                   }
                   exhausted = block_is_exhausted();
                } else {
-                  _subjective_billing.subjective_bill_failure( first_auth, trace->elapsed, fc::time_point::now() );
+                   if (!trx->read_only)
+                      _subjective_billing.subjective_bill_failure( first_auth, trace->elapsed, fc::time_point::now() );
+
                   auto e_ptr = trace->except->dynamic_copy_exception();
                   send_response( e_ptr );
                }
```

Comment:
`persist_until_expired` should be false for read-only transactions. Change 
`app().get_method<incoming::methods::transaction_async>()(pretty_input, true, true,` call in chain_plugin to be
`app().get_method<incoming::methods::transaction_async>()(pretty_input, false, true,`